From 2d4e5f8ce390a11b047bdb3f0633374a8b4b8dd2 Mon Sep 17 00:00:00 2001
From: Sandor Yu <R01008@freescale.com>
Date: Mon, 8 Dec 2014 16:01:44 +0800
Subject: [PATCH] MLK-9969 dts: Enable subdev csi driver in imx6sx AI board

Enable OV5640, VADC and CSI driver in imx6sx AI board

Signed-off-by: Sandor Yu <R01008@freescale.com>
---
 arch/arm/boot/dts/imx6sx-sabreauto.dts | 27 +++++++++++----------------
 1 file changed, 11 insertions(+), 16 deletions(-)

diff --git a/arch/arm/boot/dts/imx6sx-sabreauto.dts b/arch/arm/boot/dts/imx6sx-sabreauto.dts
index b3f9e88..1c64315 100644
--- a/arch/arm/boot/dts/imx6sx-sabreauto.dts
+++ b/arch/arm/boot/dts/imx6sx-sabreauto.dts
@@ -37,18 +37,6 @@
 		#reset-cells = <0>;
 	};
 
-	csi1_v4l2_cap {
-		compatible = "fsl,imx6sx-csi-v4l2", "fsl,imx6sl-csi-v4l2";
-		csi_id = <0>;
-		status = "okay";
-	};
-
-	csi2_v4l2_cap {
-		compatible = "fsl,imx6sx-csi-v4l2", "fsl,imx6sl-csi-v4l2";
-		csi_id = <1>;
-		status = "okay";
-	};
-
 	hannstar_cabc {
 		compatible = "hannstar,cabc";
 
@@ -473,12 +461,13 @@
 	assigned-clock-rates = <0>, <0>, <24576000>;
 };
 
-&csi1 {
-	status = "okay";
-};
-
 &csi2 {
 	status = "okay";
+	port {
+		csi2_ep: endpoint {
+			remote-endpoint = <&vadc_ep>;
+		};
+	};
 };
 
 &dcic1 {
@@ -917,4 +906,10 @@
 	vadc_in = <0>;
 	csi_id = <1>;
 	status = "okay";
+	port {
+		vadc_ep: endpoint {
+			remote-endpoint = <&csi2_ep>;
+		};
+	};
 };
+
-- 
1.8.0

